GLOSSARY.md

```diff
@@ -0,0 +1,284 @@
+If a promise has an error handler, propagation from source promises will halt
+at that promise. If a promise has many children, and only one handles errors,
+the other children will still be reported as unhandled rejections.
+
+Unwrapping `Promise<Promise<T>>` will automatically unwrap to `T`. That is:
+`.then(() => Promise.resolve(3)).then(val => console.log(val))` will
+output `3`.
+
+Internally, a promise is represented by a state machine that may be in one of
+three states — pending, resolved, or rejected. The resolved and rejected states
+are collectively referred to as "settled." Once a promise is settled, it cannot
+transition back into pending state.
+
+### Executor / Resolver
+
+The function passed to the `new Promise` constructor. Run **synchronously**.
+Not in common direct use by most promise-users — used mainly to wrap sources of
+asynchrony. As such, it rarely resolves or rejects on the same tick.
+
+Catches all errors thrown. If an error is caught, the returned promise will be
+**rejected**. Because it starts life as a rejected promise with no children, it
+will immediately hit the `OnUnhandledRejection` callback.
+
+```js
+new Promise(function Excecutor (resolve, reject) {
+  // run immediately.
+})
+```
+
+### Handler
+
+A function passed to `Promise#then` or `Promise#catch`. Receives as an argument
+the parent promise's resolved value or rejected error. Represents a new
+`Promise`. A handler which returns a value `T` or `Promise<T>` will resolve the
+new promise with `T`. A handler which throws an error `R`, or returns a
+`Promise` which itself eventually rejects (or has already rejected) with `R`,
+will reject the new promise with `R`.
+
+```js
+
+const p = new Promise(() => {})
+
+const p2 = p.then(function SuccessHandler(value) {
+  // fired if p resolves,
+}, function ErrorHandler(err) {
+  // fired if p rejects.
+})
+
+const p3 = p2.catch(function ErrorHandler(value) {
+  // fired if p2 rejects.
+})
+
+const p4 = p3.then(function SuccessHandler (value) { 
+  // resolved with p2's value if p2 resolves,
+  // resolved with p3's value if p2 rejects and p3 resolves
+})
+```
+
+### Settling
+
+The transition of a Promise from pending to a settled state, either "rejected"
+or "resolved." Once settled, a promise may never return to "pending" state.
+
+#### Rejection / reject
+
+The act of settling a promise with an error. Can be performed by the Executor
+or Handler `H` by throwing while `H` is on stack, or by resolving with a
+Promise that rejects at some point in its lifetime. In the case of Executors,
+rejection may additionally be achieved by calling the `reject` callback
+provided as an argument.
+
+#### Resolving / resolve
+
+The act of settling a promise with a value. Can be performed by the Executor or
+a Handler. Executors may only resolve by calling the `resolve` callback
+provided to them as an argument. In the case of Handlers, resolution is
+achieved by exiting the function normally _or_ by returning a value.
```

Comment:
I'm not sure that's a meaningful distinction. Exiting the function normally is returning a value (undefined).